src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/DSL.kt

```diff
@@ -15,8 +16,90 @@ import java.math.BigDecimal
 import java.time.LocalDate
 import java.time.LocalDateTime
 
-@ScopeDslMarker
-interface DataDsl {
+fun testData(repository: Repository, dsl: TestData.() -> Unit) = TestData(repository).apply(dsl)
+
+class TestData(private val repository: Repository) : TestDataDsl {
```

Comment:
The main change is to stop the Repository from implementing the top level DSL and instead having a top level class that has a reference to the Repository. Repository is already doing too much work without worrying about the DSL.